storage/range.go

```diff
@@ -964,19 +964,18 @@ func (r *Range) getLeaseForGossip(ctx context.Context) (hasLease bool, err error
 	hasLease = true
 	if !r.rm.Stopper().RunTask(func() {
 		timestamp := r.rm.Clock().Now()
-
 		// Check for or obtain the lease, if none active.
 		err = r.redirectOnOrAcquireLeaderLease(tracer.FromCtx(ctx), timestamp)
 		if err != nil {
+			hasLease = false
```

Comment:
Instead of setting hasLease to false here, can we default it to false and change it to true only when err is nil?